src/components/ModalDrawer.js

```diff
@@ -0,0 +1,131 @@
+import * as Utils from 'src/libs/utils'
+
+
+RModal.defaultStyles = { overlay: {}, content: {} }
+
+const drawer ={
+  overlay: transitionState => transitionState === 'entering' || transitionState === 'entered' ? {
+    backgroundColor: 'rgba(0, 0, 0, 0.5)', padding: '2rem 1rem',
+    display: 'flex', justifyContent: 'center', alignItems: 'flex-start',
+    position: 'fixed', left: 0, right: 0, top: 0, bottom: 0, overflowY: 'auto'
+  } : {},
+  container: state => ({
+    ...(state === 'entered' ? {} : { opacity: 0, transform: 'translate(4rem)' }),
+    top: 0, right: 0, outline: 'none',
+    transition: 'opacity 0.2s ease-out, transform 0.2s ease-out',
+    color: colors.dark(1), position: 'absolute', cursor: 'default',
+    backgroundColor: colors.light(0.4), height: '100%',
+    boxShadow: '3px 0 13px 0 rgba(0,0,0,0.3)',
+    display: 'flex', flexDirection: 'column'
+  }),
+  title: {
+    display: 'flex', alignItems: 'baseline', marginBottom: '1rem', flex: 'none', padding: '1.5rem 1.25rem'
+  },
+  body: {
+    display: 'flex', alignItems: 'baseline', marginBottom: '1rem', flex: 1, padding: '1.5rem 1.25rem', overflow: 'auto'
+  },
+  buttonRow: {
+    marginTop: 'auto', backgroundColor: colors.dark(0.2), padding: '1.75rem 1.25rem',
+    display: 'flex', justifyContent: 'flex-end', alignItems: 'baseline'
+  },
+  titleAlign: showPrevArrow => ({ marginLeft: showPrevArrow ? 'auto' : undefined })
+}
+
+const Modal = ({
+  onPrevious = () => {}, showPreviousArrow = false, transitionState = false,
+  ...props
+}) => {
+  const { onDismiss, title, titleExtras, children, width, showCancel, cancelText, showX, showButtons, okButton, ...restProps } = props
+
+  return h(RModal, {
+    parentSelector: () => document.getElementById('modal-root'),
+    isOpen: true,
+    onRequestClose: onDismiss,
+    style: { overlay: drawer.overlay(transitionState), content: { ...drawer.container(transitionState), width } },
+    ariaHideApp: false,
+    ...restProps
+  }, [
+    title && div({ style: drawer.title }, [
+      showPreviousArrow && h(Clickable, {
+        onClick: onPrevious
+      }, [icon('arrowLeft')]),
+      div({ style: { fontSize: 18, fontWeight: 600, ...drawer.titleAlign(showPreviousArrow) } }, [title]),
+      titleExtras,
+      showX && h(Clickable, {
+        style: { marginLeft: 'auto' },
+        onClick: onDismiss
+      }, [icon('times')])
+    ]),
+    div({ style: drawer.body }, [children]),
+    showButtons && div({ style: drawer.buttonRow }, [
+      showCancel ?
+        buttonSecondary({
+          style: { marginRight: '3rem' },
+          onClick: onDismiss
+        }, [cancelText]) :
+        null,
+      Utils.cond(
+        [okButton === undefined, () => buttonPrimary({ onClick: onDismiss }, 'OK')],
+        [_.isString(okButton), () => buttonPrimary({ onClick: onDismiss }, okButton)],
+        [_.isFunction(okButton), () => buttonPrimary({ onClick: okButton }, 'OK')],
+        () => okButton
```

Comment:
I don't think we should bake all this content into the base component. Even though we currently have a mockup that has this stuff, we'll likely need to support a variety of content as this evolves, and we don't really know yet what the common elements will be. Let's make this layer handle just the styling of the container and any transitions we want, and leave the content up to the consumers.

Reply:
I factored out the button and title bars